### PR TITLE
Docs: add 'gen-bounds' and 'outdated' to the list of commands.

### PR DESCRIPTION
**Pull request by danidiaz**
This commit adds entries for the 'gen-bounds' and 'outdated'
commands to the "cabal-install Commands" section of the user manual.

Formerly, those commands were only described in the "Package
description" section. Some, but not all, of the information in that
section has been moved to the entires in the list of commands.

This commit changes two other things:

- The descriptions in the "Package description" were incorrectly placed
  as sub-sections of "Opening an interpreter session". They have been
  moved one level up.

- changed the gen-bounds example, because it used dependencies which
  *already* had bounds as the starting point.

---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: doc/cabal-commands.rst [hunk 1]
```diff
@@ -403,3 +403,63 @@ the custom publishing of Haddock documentation to Hackage.
...
+
+cabal outdated
+--------------
+
+``cabal outdated [FLAGS]`` checks for outdated dependencies in the package
+description file or freeze file.
+
+``cabal outdated`` supports the following flags:
+
+- ``--freeze-file``: Read dependency version bounds from the freeze file
```
Comment (Mikolaj) [inline doc/cabal-commands.rst:424]: Perhaps call it `--v1-freeze-file` and don't mention any alias? Presumably we will soon make `--freeze-file` and alias for `--v2-freeze-file` so better start preparing users for that.

**Review by Mikolaj — APPROVED** [review]
I lazily didn't generate the documentation to check formatting and links, but otherwise LGTM.

File: doc/cabal-commands.rst (hunk 1 above)
Comment (fendor) [reply doc/cabal-commands.rst]: `--v1-freeze-file` does definitely not exist.

**Review by emilypi — APPROVED** [review]

**Comment by gbaz** [comment]
@danidiaz can you resolve the conflicts against master? Once that's done I think we should merge this. thanks!

**Review by jneira — APPROVED** [review]
Thanks for take care of docs

**Comment by gbaz** [comment]
ok i updated and added the merge flag

**Comment by emilypi** [comment]
CI stuck - merging

**Pull request merged**